KUBEVIRT_LOCAL_TESTING.md

```diff
@@ -1,34 +1,53 @@
 # [kubevirtci](README.md): Testing kubevirt with locally provisioned cluster
 
-With the changes in place you also can directly execute [`make functest`](https://github.com/kubevirt/kubevirt/blob/master/docs/getting-started.md#testing) against a cluster locally with kubevirt that was provisioned using `kubevirtci`.
+After making changes to a kubevirtci provider, you should test it locally before publishing it.
+
+With the changes in place you can also directly execute [`make functest`](https://github.com/kubevirt/kubevirt/blob/master/docs/getting-started.md#testing) against a cluster locally with kubevirt that was provisioned using `kubevirtci`.
+
+Lets go on the required steps, starting by provisioning, all the way to `make functest`.
+Note: the `...` in the beginning of `.../directory_name` means the full path of `directory_name`.
 
 Steps:
 
 ### kubevirtci: provision cluster locally
 
 ```bash
 cd .../kubevirtci
-cd cluster-provision/k8s/1.17.0
+cd cluster-provision/k8s/1.19
 ../provision.sh      # this also calls cluster-up to check whether the cluster will really start and have the pods ready
 ```
 
 ### kubevirt: prepare for tests
 
 ```bash
-# set local provision test flag
+# set local provision test flag (mandatory)
 export KUBEVIRTCI_PROVISION_CHECK=1
-# sync changes to kubevirt cluster-up, notably cluster/images.sh
+# in case testing kubevirt locally is desired, sync cluster-up folder if it has changes.
 rsync -av .../kubevirtci/cluster-up .../kubevirt/cluster-up                               
 ```                                                                                       
-                                                                                          
-### start cluster and test                                                                
+
+Note:
+In case you updated gocli and need to test it locally as well, export additionally:
+```bash
+export KUBEVIRTCI_GOCLI_CONTAINER=quay.io/kubevirtci/gocli:latest                                     
+```
+
+### start cluster (can be done in kubevirtci itself as well of course)                                                           
                                                                                           
 ```bash                                                                                   
 cd .../kubevirt                                                                           
-                                                                                          
+
+export KUBEVIRT_PROVIDER=k8s-1.19
+export KUBECONFIG=$(./cluster-up/kubeconfig.sh)
```

Comment:
earlier you asked to `cd cluster-provision/k8s/1.19` which does not have `./cluster-up/kubeconfig.sh`

Reply:
right but you have `cd .../kubevirt` line 38
so now you do
and the comment that say `(can be done in kubevirtci itself as well of course)`
which in that case it means you need to cd .../kubevirtci

is it ok ?

Reply:
but I don't really want to touch kubevirt, I only want to modify kubevirtci, and spin it up.

I understand that testing kubevirt is the main purpose of kubevirtci, but I think there should be a clear flow how to spin it up; and another addition how to sync kubevirt.

Reply:
i understand, but this is the purpose at least now of this document
and even its name
`KUBEVIRT_LOCAL_TESTING.md`

is it fine that this PR will be merged as is, as its just clarify the current flow that is presented in this PR.
and we create other md file which is kubevirtci only?

Reply:
Added KUBEVIRTCI_LOCAL_TESTING.md which is kubevirtci dedicated.